config.cc

```diff
@@ -310,24 +311,27 @@ static void logHandler(
 bool parseFile(nsjconf_t* nsjconf, const char* file) {
 	LOG_D("Parsing configuration from '%s'", file);
 
-	int fd = TEMP_FAILURE_RETRY(open(file, O_RDONLY | O_CLOEXEC));
-	if (fd == -1) {
+	google::protobuf::SetLogHandler(logHandler);
+
+	std::ifstream ifs(file);
+	if (!ifs.is_open()) {
 		PLOG_W("Couldn't open config file '%s'", file);
 		return false;
 	}
 
-	google::protobuf::SetLogHandler(logHandler);
-	google::protobuf::io::FileInputStream input(fd);
-	input.SetCloseOnDelete(true);
+	std::string conf((std::istreambuf_iterator<char>(ifs)), (std::istreambuf_iterator<char>()));
 
 	/* Use static so we can get c_str() pointers, and copy them into the nsjconf struct */
 	static nsjail::NsJailConfig nsc;
 
-	auto parser = google::protobuf::TextFormat::Parser();
-	if (!parser.Parse(&input, &nsc)) {
-		LOG_W("Couldn't parse file '%s' from Text into ProtoBuf", file);
-		return false;
+	auto status = google::protobuf::util::JsonStringToMessage(conf, &nsc);
+	if (!status.ok()) {
+		if (!google::protobuf::TextFormat::ParseFromString(conf, &nsc)) {
+			LOG_W("Couldn't parse file '%s' from Text or JSON into ProtoBuf", file);
```

Comment:
Could you test what warnings are produced by this (because of google::protobuf::SetLogHandler).

Eg. if JSON is incorrect and TextFormat correct, we probably don't want to see any warnings. Maybe SetLogHandler should save message to some tmp "static error_msg" and print it out (both json and textformat) only if both json and textformat fail?

The second problem is what you mentioned, both formats are correct. Maybe we should have 2 static nsjail::NsJailConfig variables (text_nsc and json_nsc), and if both parsing runs (json and textformat) succeed, we should print out sth like "ambiguous file format"?